syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "google/protobuf/struct.proto";

enum GrpcUserType {
    USER = 0;
    ADMIN = 1;
}

message GrpcUser {
  string id = 1;
  string username = 2;
  string email = 3;
  google.protobuf.Timestamp created_at = 4;
  oneof optional_suspended_at {
    google.protobuf.NullValue suspended_at_null = 5;
    google.protobuf.Timestamp suspended_at = 6;
  }
  GrpcUserType type = 7;
}
